pubsub/azure/eventhubs/eventhubs.go

```diff
@@ -98,62 +117,304 @@ func subscribeHandler(ctx context.Context, topic string, e *eventhub.Event, hand
+		return fmt.Errorf("error creating event hub %s: %s", hubName, err)
+	}
+	return nil
+}
 
-	if val, ok := meta.Properties[consumerID]; ok && val != "" {
-		m.consumerGroup = val
+func (aeh *AzureEventHubs) ensurePublisherClient(hubName string) error {
+	if aeh.metadata.EnableEnitiyManagement {
+		if err := aeh.ensureEventHub(hubName); err != nil {
+			return err
+		}
+	}
+	userAgent := "dapr-" + logger.DaprVersion
+	if aeh.metadata.ConnectionString != "" {
+		// Connect with connection string.
+		newConnectionString, err := aeh.constructConnectionStringFromTopic(hubName)
+		if err != nil {
+			return err
+		}
+
+		hub, err := eventhub.NewHubFromConnectionString(newConnectionString,
+			eventhub.HubWithUserAgent(userAgent))
+		if err != nil {
+			aeh.logger.Debugf("unable to connect to azure event hubs: %v", err)
+			return fmt.Errorf("unable to connect to azure event hubs: %v", err)
+		}
+		aeh.hubClients[hubName] = hub
 	} else {
-		return m, errors.New(missingConsumerIDErrorMsg)
+		if hubName == "" {
+			return errors.New("error: missing topic/hubName attribute with AAD connection")
+		}
+
+		hub, err := eventhub.NewHub(aeh.metadata.EventHubNamespace, hubName, aeh.tokenProvider, eventhub.HubWithUserAgent(userAgent))
+		if err != nil {
+			return fmt.Errorf("unable to connect to azure event hubs: %v", err)
+		}
+		aeh.hubClients[hubName] = hub
+	}
+
+	return nil
+}
+
+func (aeh *AzureEventHubs) ensureSubscriberClient(topic string, leaserCheckpointer *storage.LeaserCheckpointer) (*eph.EventProcessorHost, error) {
+	// connectionString given.
+	if aeh.metadata.ConnectionString != "" {
+		hubName, err := validateAndGetHubName(aeh.metadata.ConnectionString)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing connection string %s", err)
+		}
+		if hubName != "" && hubName != topic {
+			return nil, fmt.Errorf("error: component cannot subscribe to requested topic %s with the given connectionString", topic)
+		}
+		if hubName == "" {
+			aeh.logger.Debugf("eventhub namespace connection string given. using topic as event hub entity path")
+		}
+		connectionString, err := aeh.constructConnectionStringFromTopic(topic)
+		if err != nil {
+			return nil, err
+		}
+		processor, err := eph.NewFromConnectionString(aeh.ctx, connectionString, leaserCheckpointer, leaserCheckpointer, eph.WithNoBanner(), eph.WithConsumerGroup(aeh.metadata.ConsumerGroup))
+		if err != nil {
+			return nil, err
+		}
+		aeh.logger.Debugf("processor initialized via connection string for topic %s", topic)
+		return processor, nil
+	}
+	// AAD connection.
+	processor, err := eph.New(aeh.ctx, aeh.metadata.EventHubNamespace, topic, aeh.tokenProvider, leaserCheckpointer, leaserCheckpointer, eph.WithNoBanner(), eph.WithConsumerGroup(aeh.metadata.ConsumerGroup))
+	if err != nil {
+		return nil, err
+	}
+	aeh.logger.Debugf("processor initialized via AAD for topic %s", topic)
+
+	return processor, nil
+}
+
+func (aeh *AzureEventHubs) createHubManager() error {
```

Comment:
HubManager client from azure-event-hubs go sdk for creating/managing an event hub.